watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1175 @@
 _logger.addHandler(handler)
 _logger.setLevel(logging.DEBUG)
 
-# TODO: consider config for file_name for each writing method
-
 
 class Flash:
     """
-    A class to execute OLI Cloud flash calculations, replacing and augmenting WaterAnalysis class functionality.
-
-    :param water_analysis_properties: dictionary containing pre-built water analysis input template blocks
-    :param optional_properties: dictionary containing pre-configured optional properties to attach to OLI calls (defaults to True for all properties)
-    :param input_unit_set: dictionary containing conversions between OLI and Pyomo unit names
-    :param output_unit_set: dictionary containing preferred units for output expression
-    :param stream_output_options: dictionary pointing to properties that can be extracted from flash stream outputs
-    :param relative_inflows: bool to switch between additive and absolute substitutions for solutes when using surveys
-    :param interactive_mode: bool to switch level of logging display from info to debug only
+    A class to execute OLI Cloud flash calculations.
+
+    :param optional_properties: dictionary for optional properties to attach to OLI calls, all True by default
+    :param input_unit_set: dictionary for conversions between OLI and Pyomo unit names
+    :param output_unit_set: dictionary for preferred output units
+    :param relative_inflows: bool switch for surveys - true to add specified value to initial value, false to replace initial value with specified value
+    :param debug_level: string defining level of logging activity
     """
 
     def __init__(
         self,
-        water_analysis_properties=water_analysis_properties,
         optional_properties=optional_properties,
         input_unit_set=input_unit_set,
         output_unit_set=output_unit_set,
-        stream_output_options=stream_output_options,
         relative_inflows=True,
-        interactive_mode=True,
+        debug_level="INFO",
     ):
-        # set values based on inputs
-        self.water_analysis_properties = water_analysis_properties
         self.optional_properties = optional_properties
         self.input_unit_set = input_unit_set
         self.output_unit_set = output_unit_set
-        self.stream_output_options = stream_output_options
         self.relative_inflows = relative_inflows
```

Comment:
is there any case where the user may want to do different types relative vs absolute) for different survey variables? If that is the case, should we be thinking about making this a multi-dimensional vector where each survey has its own Boolean?

Just a thought.

Reply:
Yes, probably for pH, some state variables that are usually within certain ranges, etc. I'll add it to the issue #1328